src/main/java/org/spongepowered/api/item/data/FireworkRocket.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.data;
+
+import org.spongepowered.api.item.meta.FireworkExplosion;
+
+import java.util.Set;
+
+/**
+ * Represents a firework rocket in an inventory.
+ */
+public interface FireworkRocket extends ItemData {
+
+    /**
+     * Gets the explosions this firework will cause. Not read-only.
+     * 
+     * @return The explosions this firework will cause
+     */
+    Set<FireworkExplosion> getExplosions();
```

Comment:
Immutability is beautiful in the sense that the collection *returned* can not be changed, but by adding `add(Foo)` and `remove(Bar)`, we can guarantee we are the only ones mutating the internal collection. If you give out references to the collection, many things can happen in regards to changing the collection, possibly during serialization, concurrent modifications, etc.